src/ir/daphneir/Canonicalize.cpp

```diff
@@ -249,6 +249,25 @@ mlir::LogicalResult mlir::daphne::SparsityOp::canonicalize(mlir::daphne::Sparsit
 mlir::LogicalResult mlir::daphne::EwAddOp::canonicalize(mlir::daphne::EwAddOp op, PatternRewriter &rewriter) {
     mlir::Value lhs = op.getLhs();
     mlir::Value rhs = op.getRhs();
+    // This will check for the fill operation on the left hand side to push down the arithmetic inside
+    // of it
+    auto lhsFill = lhs.getDefiningOp<mlir::daphne::FillOp>();
+    if (lhsFill) {
+        auto lhsOp = lhs.getDefiningOp();
+        auto fillValue = lhsOp->getOperand(0);
```

Comment:
Accessing the arguments of the `FillOp` by their name can make the code more readable, e.g., `auto fillValue = lhsFill.getArg();`. You can find the names of the arguments in `src/ir/daphneir/DaphneOps.td`.

Reply:
I am not able to find `getArg` in https://mlir.llvm.org/doxygen/classmlir_1_1Operation.html .

```
/daphne/src/ir/daphneir/Canonicalize.cpp:259:29: error: 'class mlir::Operation' has no member named 'getArg'
  259 |         auto width = lhsOp->getArg("numCols");
```

Reply:
> Thanks, let me double check my LSP setup.
> 
> Unfortunately I still get the same error during compilation though:
> 

It's not a pointer.

I only made this change in your branch and this works on my side.

```diff
-        auto fillValue = lhsFill->getOperand(0);
+        auto fillValue = lhsFill.getArg();
         auto height = lhsFill->getOperand(1);

```
I'd also declare it like so, `auto lhsFill = ..` as `clang-tidy` complains (rule `modernize-use-auto`).